docs/DeveloperGuide.md

```diff
@@ -3,22 +3,30 @@ layout: page
 title: Developer Guide
 ---
 * Table of Contents
-{:toc}
+- [Acknowledgements](#acknowledgements)
+- [Setting Up](#setting-up)
+- [Design](#design)
+- [Implementation](#implementation)
+- [Misc. Docs](#misc-documentation)
+- [Appendix](#appendix)
```

Comment:
I think the {:toc} automatically generates the table of contents with jekyll, as you can see from the existing site, where the TOC is inserted. 
https://ay2324s2-cs2103t-w12-2.github.io/tp/DeveloperGuide.html, so I don't think this needs to be done.

Reply:
The idea here was to have finer control over nesting without having to change heading levels (`#`, `##` etc.). We can always revert this later if it ends up being something we won't use.